galasa-managers-parent/galasa-managers-cicsts-parent/dev.galasa.cicsts.manager/src/main/java/dev/galasa/cicsts/ITsqFactory.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package dev.galasa.cicsts;
+
+import javax.validation.constraints.NotNull;
+public interface ITsqFactory {
+
+      /**
+       * Create a new ITsq object with recoverable status   
+       * @param queueName TSQ name
+       * @param recoverable true for recoverable and false for non-recoverable
+       * @return ITsq object
+       * @throws TsqException if there is a problem in creating the ITsq object
+       */
+      public ITsq createQueue(@NotNull String queueName, @NotNull boolean recoverable) throws TsqException;
```

Comment:
I think we need a ModelFactory where you can create an ICicsModel or something similar.
eg: 
```
interface ICicsModelFactory {
  public IModel createModel( String namePrefix, boolean isRecoverable, isShared...etc)
}
```

Then in this ITsqFactory interface, we can support:
```
/** For those queues which don't want or need a model */
public ITsq createQueue(@NotNull String queueName);

And we could also add
public ITsq createQueue( ICicsModel model, String queueNameSuffix );
```
If we did that, then you would get the 'isRecoverable' from the model explicitly, rather than implicitly via name-prefix matching.

Reply:
Hi @techcobweb , TSModel is a CICS resource.  As we discussed in the meeting last week, I will be adding a new method in CICSResourceManager for creating TSModel.  However as advised by you, that will go as a separate change after TSQ manager.  TSQ manager is specific for TSQs only.